lstchain/io/event_selection.py

```diff
@@ -141,6 +145,35 @@ def apply_global_gh_cut(self, data):
         """
         return data[data["gh_score"] > self.global_gh_cut]
 
+    def update_fill_cuts(self, cut_table):
+        """
+        For an energy-dependent cut table, update the cuts for bins with number
+        of events fewer than the minimum number, for which pyirf uses a
+        constant fill_value, usually at the energy threshold limits, with cut
+        evaluated at the nearest bin with number of events more than the given
+        minimum.
+        """
+        cut_table_new = cut_table.copy()
+
+        low_event_bins = np.argwhere(
+            cut_table["n_events"] < self.min_event_p_en_bin
+        ).ravel()
+        high_event_bins = np.where(
+            cut_table["n_events"] >= self.min_event_p_en_bin
+        )[0]
+
+        for low_ in low_event_bins:
+            if low_ < high_event_bins[0]:
+                cut_table_new["cut"].value[low_] = cut_table["cut"].value[
+                    high_event_bins[0]
+                ]
+            elif low_ > high_event_bins[-1]:
+                cut_table_new["cut"].value[low_] = cut_table["cut"].value[
+                    high_event_bins[-1]
+                ]
```

Comment:
This does not consider the possibility (perhaps unlikely) that you have a bin without enough statistics which has higher energy than a bin with enough statistics (or the corresponding case in the high-E end of the range). In that case an interpolation could be used.

Reply:
Added coverage for this possibility and updated the test accordingly.

Reply:
I think, in that even more unlikely case, we/analyzer have provided a very large value for `min_event_p_en_bin` with respect to the simulation data that we have, and we can tune that instead of covering for this computational possibility.